Update CL functions docs: removal of inline secrets (#1229)
smartcontractkit/documentation: src/pages/chainlink-functions/tutorials/api-use-secrets.mdx

import { Aside } from "@components"
import ChainlinkFunctions from "@features/chainlink-functions/common/ChainlinkFunctions.astro"

This tutorial shows you how to send a request to a Decentralized Oracle Network to call the [Coinmarketcap API](https://coinmarketcap.com/api/documentation/v1/). After [OCR](/chainlink-functions/resources/concepts/) completes off-chain computation and aggregation, it returns the `BTC/USD` asset price to your smart contract. Because the API requires you to provide an API key, this guide will also show you how to encrypt, sign your API key, and share it off-chain with a Decentralized Oracle Network (DON).

> The Chainlink Functions starter kit creates [gists](https://docs.github.com/en/get-started/writing-on-github/editing-and-sharing-content-with-gists/creating-gists) containing encrypted secrets on your behalf, then share the gist URL with the DON. This method comes with a security benefit:

Reviewer: ```suggestion
The Chainlink Functions starter kit creates [gists](https://docs.github.com/en/get-started/writing-on-github/editing-and-sharing-content-with-gists/creating-gists) containing encrypted secrets on your behalf. Then, it shares the gist URL with the DON. This method comes with a security benefit:
```

> - The encrypted secrets are never stored on-chain. The secrets are encrypted with the DON's public key so that only an oracle node in the DON can decrypt them using the DON's private key. After the DON fulfills a request, the starter kit deletes the gist. Which allows to revoking credentials once the DON has fulfilled the request.

Reviewer: ```suggestion
- The encrypted secrets are never stored on-chain. The secrets are encrypted with the DON's public key so that only an oracle node in the DON can decrypt them using the DON's private key. After the DON fulfills a request, the starter kit deletes the gist. This revokes the credentials after the DON has fulfilled the request.
```

   ```

1. Get a free API key from [CoinMarketCap](https://coinmarketcap.com/api/).
1. Run `npx env-enc` to add an encrypted `COINMARKETCAP_API_KEY` to your `.env.enc` file.

   ```shell
   npx env-enc set COINMARKETCAP_API_KEY "YOUR_COINMARKETCAP_API_KEY"
   ```

> 1. The starter kit will store the encrypted secrets as [gists](https://docs.github.com/en/get-started/writing-on-github/editing-and-sharing-content-with-gists/creating-gists) to share them off-chain with the Decentralized Oracle Network. To allow the starter kit to write gists on your behalf, create a [github fine-grained personal access token](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token#creating-a-fine-grained-personal-access-token).

Reviewer: ```suggestion
1. The starter kit store encrypted secrets as [gists](https://docs.github.com/en/get-started/writing-on-github/editing-and-sharing-content-with-gists/creating-gists) to share them off-chain with the Decentralized Oracle Network. To allow the starter kit to write gists on your behalf, create a [github fine-grained personal access token](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token#creating-a-fine-grained-personal-access-token).
```
